cf-execd/cf-execd-runagent.c

```diff
@@ -30,17 +30,70 @@
 #include <pipes.h>
 #include <alloc.h>
 #include <file_lib.h>
+#include <string_lib.h>
 #include <known_dirs.h>
 
 #include <cf-execd-runagent.h>
 
+static inline bool StringIsLocalhostIP(const char *str)
+{
+    assert(str != NULL);
+
+    const char *c = str;
+    if (*(c++) != '1')
+    {
+        return false;
+    }
+    if (*(c++) != '2')
+    {
+        return false;
+    }
+    if (*(c++) != '7')
+    {
+        return false;
+    }
+    if (*(c++) != '.')
+    {
+        return false;
+    }
+    while ((*c != '\0') && (*c != '.'))
+    {
+        if (!isdigit(*(c++)))
+        {
+            return false;
+        }
+    }
+    if (*(c++) != '.')
+    {
+        return false;
+    }
+    while ((*c != '\0') && (*c != '.'))
+    {
+        if (!isdigit(*(c++)))
+        {
+            return false;
+        }
+    }
+    if (*(c++) != '.')
+    {
+        return false;
+    }
+    while (*c != '\0')
+    {
+        if (!isdigit(*(c++)))
+        {
+            return false;
+        }
+    }
+    return true;
```

Comment:
Seems like sscanf would be easier. This function would accept `127.999999.999999.999999` as a localhost IP.A unit test would help explain the behavior of this function.

Reply:
This function was just supposed to be a quick and cheap helper recognition, not validation, mechanism.  I'll add the digit count checks, but I don't really want to go full road here with a real validation function.

Reply:
Task for a proper implementation: https://tracker.mender.io/browse/CFE-3677